pkg/operation/botanist/controlplane.go

```diff
@@ -1302,43 +1304,31 @@ func (b *Botanist) DeployKubeControllerManager(ctx context.Context) error {
 	return b.K8sSeedClient.ChartApplier().Apply(ctx, filepath.Join(chartPathControlPlane, v1beta1constants.DeploymentNameKubeControllerManager), b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameKubeControllerManager, kubernetes.Values(values))
 }
 
-// DeployKubeScheduler deploys kube-scheduler deployment.
-func (b *Botanist) DeployKubeScheduler(ctx context.Context) error {
-	defaultValues := map[string]interface{}{
-		"replicas":          b.Shoot.GetReplicas(1),
-		"kubernetesVersion": b.Shoot.Info.Spec.Kubernetes.Version,
-		"podAnnotations": map[string]interface{}{
-			"checksum/secret-kube-scheduler":        b.CheckSums[v1beta1constants.DeploymentNameKubeScheduler],
-			"checksum/secret-kube-scheduler-server": b.CheckSums[common.KubeSchedulerServerName],
-		},
-		"podLabels": map[string]interface{}{
-			v1beta1constants.LabelPodMaintenanceRestart: "true",
-		},
-	}
-
-	if b.ShootedSeed != nil {
-		defaultValues["resources"] = map[string]interface{}{
-			"limits": map[string]interface{}{
-				"cpu":    "300m",
-				"memory": "512Mi",
-			},
-		}
+// DefaultKubeScheduler returns a deployer for the kube-scheduler.
+func (b *Botanist) DefaultKubeScheduler() (kubescheduler.KubeScheduler, error) {
+	image, err := b.ImageVector.FindImage(common.KubeSchedulerImageName, imagevector.RuntimeVersion(b.SeedVersion()), imagevector.TargetVersion(b.ShootVersion()))
+	if err != nil {
+		return nil, err
 	}
 
-	if schedulerConfig := b.Shoot.Info.Spec.Kubernetes.KubeScheduler; schedulerConfig != nil {
-		defaultValues["featureGates"] = schedulerConfig.FeatureGates
-
-		if schedulerConfig.KubeMaxPDVols != nil {
-			defaultValues["kubeMaxPDVols"] = *schedulerConfig.KubeMaxPDVols
-		}
-	}
+	return kubescheduler.New(
+		b.K8sSeedClient.DirectClient(),
+		b.Shoot.SeedNamespace,
+		b.Shoot.Info.Spec.Kubernetes.Version,
+		image.String(),
+		int32(b.Shoot.GetReplicas(1)),
+		b.Shoot.Info.Spec.Kubernetes.KubeScheduler,
+	)
+}
 
-	values, err := b.InjectSeedShootImages(defaultValues, common.KubeSchedulerImageName)
-	if err != nil {
-		return err
-	}
+// DeployKubeScheduler deploys the Kubernetes scehduler.
+func (b *Botanist) DeployKubeScheduler(ctx context.Context) error {
+	b.Shoot.Components.ControlPlane.KubeScheduler.SetSecrets(kubescheduler.Secrets{
+		Kubeconfig: kubescheduler.Secret{Name: "kube-scheduler", Checksum: b.CheckSums[v1beta1constants.DeploymentNameKubeScheduler]},
+		Server:     kubescheduler.Secret{Name: "kube-scheduler-server", Checksum: b.CheckSums[common.KubeSchedulerServerName]},
```

Comment:
Can we have a separate constant for the name of the kubeconfig secret to avoid confusion/mistakes?
Maybe it makes sense to have those constants in the kubescheduler package or should we have them in the common package?

Also, can you please pass the respective constants to the `Name` field, as it will be used as the `SecretRef` in the Volume?
```suggestion
		Kubeconfig: kubescheduler.Secret{Name: v1beta1constants.SecretNameKubeScheduler Checksum: b.CheckSums[v1beta1constants.SecretNameKubeScheduler]},
		Server:     kubescheduler.Secret{Name: common.KubeSchedulerServerName, Checksum: b.CheckSums[common.KubeSchedulerServerName]},
```